staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -569,6 +569,27 @@ func isExpiredError(err error) bool {
 	return apierrors.IsResourceExpired(err) || apierrors.IsGone(err)
 }
 
+var tooLargeResourceVersionCauseMsg = "Too large resource version"
+
 func isTooLargeResourceVersionError(err error) bool {
-	return apierrors.HasStatusCause(err, metav1.CauseTypeResourceVersionTooLarge)
+	if apierrors.HasStatusCause(err, metav1.CauseTypeResourceVersionTooLarge) {
+		return true
+	}
+	// In Kubernetes 1.18.5 and earlier, the api server doesn't set status cause
+	// metav1.CauseTypeResourceVersionTooLarge when returning such error. In order
+	// to ensure backwards compatibility with these server versions we must check if
+	// the message field of the timeout error matches the known string.
+	if !apierrors.IsTimeout(err) {
+		return false
+	}
+	apierr, ok := err.(apierrors.APIStatus)
+	if !ok || apierr == nil || apierr.Status().Details == nil {
+		return false
+	}
+	for _, cause := range apierr.Status().Details.Causes {
+		if cause.Message == tooLargeResourceVersionCauseMsg {
```

Comment:
should we make this an inline string with a comment that it matches the message returned by 1.18.0-1.18.5 and should not be changed? referencing a variable seems vulnerable to accidental change

Reply:
makes sense, especially in the unit test.